framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/KieServerDeploymentImpl.java

```diff
@@ -83,6 +92,41 @@ public String getSecureServiceName() {
         super.waitForScale();
         if (getInstances().size() > 0) {
             RouterUtil.waitForRouter(getUrl());
+            waitForSecureRouter();
+        }
+    }
+
+    private void waitForSecureRouter() {
+        LocalDateTime endTime = LocalDateTime.now().plus(Duration.of(5, ChronoUnit.SECONDS));
+        while (LocalDateTime.now().isBefore(endTime)) {
```

Comment:
Not sure why you have this loop, because the method sendSecureRequiest() will throw and exception if non-success response is returned, so the loop will always run just once.